#! /bin/sh /usr/share/dpatch/dpatch-run
## opt-49-x_pvrinput.dpatch by Tobias Grimm <tg@e-tobi.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: This patch is required by the pvrinput plugin and was taken from
## DP: version 2007-11-29 of the plugin.
## DP: http://drseltsam.device.name/vdr/pvr/src/pvrinput
## DP:
## DP: 2008-05-14: Added CA-Id's 0xA1 and 0xA2, required by pvrusb2
## DP:             (as suggested by Christian Jarczyk)

@DPATCH@
diff -urNad vdr-1.5.15~/dvbdevice.c vdr-1.5.15/dvbdevice.c
--- vdr-1.5.15~/dvbdevice.c	2008-02-09 17:11:44.000000000 +0100
+++ vdr-1.5.15/dvbdevice.c	2008-02-20 14:38:36.000000000 +0100
@@ -759,7 +759,7 @@
   bool result = false;
   bool hasPriority = Priority < 0 || Priority > this->Priority();
   bool needsDetachReceivers = false;
-
+  if (Channel->Ca(0) == 0xA0 || Channel->Ca(0) == 0xA1 || Channel->Ca(0) == 0xA2) return false; /* this caid marks a pvrinput plugin channel, no dvb */
   if (ProvidesSource(Channel->Source())) {
      result = hasPriority;
      if (Priority >= 0 && Receiving(true)) {
